jwst/associations/tests/test_pool.py

```diff
@@ -1,15 +1,16 @@
+import os
 from jwst.associations.tests.helpers import t_path
-
 from jwst.associations import AssociationPool
 
 POOL_FILE = t_path('data/jw93060_20150312T160130_pool.csv')
 
 
-def test_pool(tmpdir):
+def test_pool(tmp_path_factory):
     pool = AssociationPool.read(POOL_FILE)
     assert len(pool) == 636
 
-    tmp_pool = str(tmpdir.mkdir(__name__).join('tmp_pool.csv'))
+    tmp_path = tmp_path_factory.mktemp(__name__)
+    tmp_pool = os.path.join(tmp_path, 'tmp_pool.csv')
```

Comment:
@jdavies-st Curious as to why you think your suggested change is better than using `os.path.join`?  Is the explicit use of "/" as generic as letting `join` figure out the proper syntax for you?

Reply:
I guess the reason is that the "/" syntax returns a path type, while join converts to a string.  I'll change it to the suggestion of @jdavies-st  and see what happens

Reply:
Surprisingly, making this change did cause several failures.  I'm going to go ahead and push an intermediate commit so y'all can see the tracebacks, even though I'm not done with all of the changes.

One example is in test_ramp_fit_step.py.  Changing `gainfile` and `readnoisefile` to be paths causes `stpipe.config_parser.ValidationError: Config parameter 'override_gain': the value "/private/var/folders/4c/gd84fq3972vc4jnbtqmnrrnr0005r1/T/pytest-of-emolter/pytest-229/data0/gain.fits" is of the wrong type.`. The pipeline step is called using the `.Call()` method, so everything seems Pythonic, and if I were a user, I would probably expect passing a path should work fine.

It could be that my changes are not quite right - maybe these tests need to be passed the `tmp_cwd` fixture or something - but if it is correct, what is your perspective on allowing the validator to accept Path objects in stpipe?

Reply:
Yeah, `stpipe.Step` parameter validation is a little old-fashioned.  It requires either a string or open `datamodel` passed to these.  I would suggest converting these paths to strings for these specific cases.  I can open an issue in `stpipe` to make sure that any attribute that accepts a string path and validates it as a string should also accept a `pathlib.Path` object and validate it the same way.  I'll link it here.

Reply:
> @jdavies-st Curious as to why you think your suggested change is better than using `os.path.join`? Is the explicit use of "/" as generic as letting `join` figure out the proper syntax for you?

As @emolter said, `os.path.join` returns a string, and it works.  But if we would like to make sure we catch instances where the runtime code in `jwst` explicitly assumes a path is a string and therefore not compatible with `pathlib.Path` objects, then it would be good to make sure any paths we pass to our runtime code in tests is a `pathlib.Path` object.

As for `/`, it's just the operator shorthand for joining `Path` parts using the `Path.join_path()` method.  See https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.joinpath and https://docs.python.org/3/library/pathlib.html#basic-use

See https://docs.python.org/3/library/pathlib.html#correspondence-to-tools-in-the-os-module for correspondence with `os.path` functions.

Reply:
Oh, so the "/" doesn't get used literally? I was wondering because of course for different OS's (like the dreaded Windows) it's different - a backslash.

Reply:
Correct.  It's an overloaded python operator.  And it's very readable in code, compared to use the methods `.join_path()` or `os.path.join`.

Reply:
thanks for opening that issue @jdavies-st , for now I will work around it as you suggest